test.js

```diff
@@ -1,46 +1,54 @@
-'use strict';
-var assert = require('assert');
-var gutil = require('gulp-util');
-var through2 = require('through2');
-var jasmine = require('./');
-var out = process.stdout.write.bind(process.stdout);
-
-it('should run unit test and pass', function (cb) {
-	var stream = jasmine({
-		timeout: 9000,
-		verbose: true
-	});
+import test from 'ava';
+import gutil from 'gulp-util';
+import through2 from 'through2';
+import fn from './';
 
-	process.stdout.write = function (str) {
-		out(str);
+const out = process.stdout.write.bind(process.stdout);
 
-		if (/should pass/.test(str)) {
-			assert(true);
-			process.stdout.write = out;
-			cb();
-		}
-	};
+function jasmine(options) {
+	return new Promise((resolve, reject) => {
+		const stream = fn(options);
 
-	stream.write(new gutil.File({
-		path: 'fixture.js',
-		contents: new Buffer('')
-	}));
+		let output = '';
 
-	stream.end();
-});
+		process.stdout.write = function (str) {
+			out(str);
 
-it('should run the test only once even if called in succession', function (done) {
-	var stream = jasmine({
-		timeout: 9000,
-		verbose: true
+			output += str;
+		};
+
+		stream.on('data', () => { });
+
+		stream.on('error', reject);
+
+		stream.on('end', () => {
+			resolve(output);
+		});
+
+		stream.write(new gutil.File({
+			path: 'fixture.js',
+			contents: new Buffer('')
+		}));
+
+		stream.end();
 	});
-	var output = '';
-	var reader = through2.obj(function (file, enc, cb) {
+}
+
+test('run unit test and pass', async t => {
+	const stdout = await jasmine({timeout: 9000, verbose: true});
```

Comment:
Created a separate function because I hoped I could use it in the second test as well. It could be used to add more tests in the future. If you don't care, I can roll it back and do everything inline.